src/Upload.js

```diff
@@ -69,25 +70,35 @@ export default class Upload {
       const localResumeIndex = meta.getResumeIndex()
       const remoteResumeIndex = await getRemoteResumeIndex()
 
-      const resumeIndex = Math.min(localResumeIndex, remoteResumeIndex)
-      this.debug(`Validating chunks up to index ${resumeIndex}`)
-      this.debug(` - Remote index: ${remoteResumeIndex}`)
-      this.debug(` - Local index: ${localResumeIndex}`)
-
-      try {
-        await processor.run(validateChunk, 0, resumeIndex)
-      } catch (e) {
-        this.debug('Validation failed, starting from scratch')
-        this.debug(` - Failed chunk index: ${e.chunkIndex}`)
-        this.debug(` - Old checksum: ${e.originalChecksum}`)
-        this.debug(` - New checksum: ${e.newChecksum}`)
-
-        await processor.run(uploadChunk)
-        return
+      let resumeIndex = remoteResumeIndex;
+      if(opts.validateChecksum) {
+        resumeIndex = Math.min(localResumeIndex, remoteResumeIndex)
+        this.debug(`Validating chunks up to index ${resumeIndex}`)
+        this.debug(` - Local index: ${localResumeIndex}`)
       }
 
-      this.debug('Validation passed, resuming upload')
-      await processor.run(uploadChunk, resumeIndex)
+      this.debug(` - Remote index: ${remoteResumeIndex}`)
+
+      if (opts.validateChecksum) {
+        try {
+          await processor.run(validateChunk, 0, resumeIndex)
+        } catch (e) {
+          this.debug('Validation failed, starting from scratch')
```

Comment:
Is there a way to notify the user of this error?

Reply:
I don't think there's a need for it. This is the default behavior of the library, if the validation fails it simply starts the process from scratch. But with this PR I'm bypassing this validation, as it breaks with large files, and I don't think we need it.

Reply:
Got it 👍